crates/bevy_math/src/compass.rs

```diff
@@ -170,6 +214,53 @@ impl CompassOctant {
             Self::NorthWest => Self::SouthEast,
         }
     }
+
+    /// Checks if a point is in the direction represented by this [`CompassOctant`] from an origin.
+    ///
+    /// This uses a cone-based check: the vector from origin to the candidate point
+    /// must have a positive dot product with the direction vector.
+    ///
+    /// **Note**: This function assumes **UI coordinates** where Y increases downward.
+    /// This is the opposite of mathematical coordinates where Y increases upward.
```

Comment:
I think reversing the y-axis direction for some of the methods of `CompassOctant` is going to be really confusing. Just adding a doc-comment isn't sufficient, expecting the directions to be consistant is very natural.